### PR TITLE
Support manual DHT record insertion

### PR DESCRIPTION
**Pull request by dmitry-markin**
Support manual DHT record insertion to comply with `libp2p` implementation.

### CODE REVIEW

File: src/protocol/libp2p/kademlia/store.rs [hunk 1]
```diff
@@ -25,6 +25,9 @@ use crate::protocol::libp2p::kademlia::record::{Key, Record};
 
 use std::collections::{hash_map::Entry, HashMap};
 
+/// Logging target for the file.
+const LOG_TARGET: &str = "litep2p::ipfs::store";
```
Comment (lexnv) [inline src/protocol/libp2p/kademlia/store.rs:29]: dq: Are we using ipfs in other places? Does it make sens for us to use `kad` here instead?

File: tests/protocol/kademlia.rs [hunk 2]
```diff
@@ -122,3 +125,312 @@ async fn put_value() {
...
+                panic!("record was not stored in 10 secs")
+            }
+            _ = litep2p1.next_event() => {}
+            _ = litep2p2.next_event() => {}
+            _ = kad_handle1.next() => {}
+            event = kad_handle2.next() => {
+                match event {
+                    Some(KademliaEvent::IncomingRecord { record }) => {
+                        // Check if the record was stored.
+                        let _ = kad_handle2
```
Comment (lexnv) [inline tests/protocol/kademlia.rs:176]: nit; We could probably compare the records here

File: src/protocol/libp2p/kademlia/store.rs (hunk 1 above)
Comment (dmitry-markin) [reply src/protocol/libp2p/kademlia/store.rs:29]: Yes, we are using `litep2p::ipfs::kademlia` in other places. Corrected.

File: tests/protocol/kademlia.rs (hunk 2 above)
Comment (dmitry-markin) [reply tests/protocol/kademlia.rs:177]: Thanks, done.

Comment (lexnv) [reply tests/protocol/kademlia.rs:177]: Perfect, thanks!

**Review by lexnv — APPROVED** [review]
LGTM! Nice work here!

**Review by alexggh — APPROVED** [review]

**Pull request merged**